sound/soc/sof/intel/hda.c

```diff
@@ -185,6 +185,10 @@ static int hda_sdw_probe(struct snd_sof_dev *sdev)
 
 	memset(&res, 0, sizeof(res));
 
+	/* sdw trigger callback is only needed for IPC4 */
+	if (sdev->pdata->ipc_type == SOF_INTEL_IPC4)
+		sdw_callback.trigger = ipc4_sdw_trigger;
```

Comment:
@bardliao would it be possible to roll this change into hda_set_dai_drv_ops(). It might man moving the definition for the struct sdw_intel_ops sdw_callback as well but it will keep all the IPC4 DAI related ops settings in one place

Reply:
> @bardliao would it be possible to roll this change into hda_set_dai_drv_ops(). It might mean moving the definition for the struct sdw_intel_ops sdw_callback as well but it will keep all the IPC4 DAI related ops settings in one place

No, hda_set_dai_drv_ops() can only set ops for those dai drivers in ops->drv[]. But sdw dai driver is created in `drivers/soundwire/intel.c`, and that's why we need a new .trigger callback in `struct sdw_intel_ops {}`